clientv3/balancer.go

```diff
@@ -26,32 +27,89 @@ import (
 // simpleBalancer does the bare minimum to expose multiple eps
 // to the grpc reconnection code path
 type simpleBalancer struct {
-	// eps are the client's endpoints stripped of any URL scheme
-	eps     []string
-	ch      chan []grpc.Address
-	numGets uint32
+	// addrs are the client's endpoints for grpc
+	addrs []grpc.Address
+	// notifyCh notifies grpc of the set of addresses for connecting
+	notifyCh chan []grpc.Address
+
+	// upc receives a message whenever up is posted
+	upc chan struct{}
+
+	// mu protects upEps, pinAddr, and connAddr
+	mu sync.Mutex
+	// upEps holds the current endpoints that have an active connection
+	upEps map[string]struct{}
+	// pinAddr is the currently pinned connected address
+	pinAddr string
+	// connAddr is the address proposed for the next connection
+	connAddr string
```

Comment:
connetctingAddr? Once we return this to gRPC, grpc will start to connecting this addr. Right?